runtime/runtime/src/balance_checker.rs

```diff
@@ -43,20 +45,20 @@ pub(crate) fn check_balance(
         } else {
             0
         };
-    let total_accounts_balance = |state| -> Result<Balance, StorageError> {
+    let total_accounts_balance = |state| -> Result<Balance, RuntimeError> {
         Ok(all_accounts_ids
             .iter()
             .map(
                 |account_id| Ok(get_account(state, account_id)?.map_or(0, |a| a.amount + a.locked)),
             )
             .collect::<Result<Vec<Balance>, StorageError>>()?
             .into_iter()
-            .sum::<Balance>())
+            .try_fold(0u128, |res, balance| safe_add_balance(res, balance))?)
```

Comment:
This would return `RuntimeError::InvalidTxError::CostOverflow`, but we need to throw `RuntimeError:: UnexpectedIntegerOverflow`

Reply:
> This would return RuntimeError::InvalidTxError::CostOverflow

why?

Reply:
Where does IntegerOverflowError converts into RuntimeError::UnexpectedIntegerOverflow in the code?
Add a test to check that it actually does return RuntimeError::UnexpectedIntegerOverflow

Reply:
Sorry, my bad. I forgot to add this specific conversion. Take a look on the updated code.